### PR TITLE
#419: feat: add support for command [JSON.DEL]

### PR DESCRIPTION
**Pull request by jujiale**
hello:
this pr is link issue: #419 

- add json.del command. refer to redis [json.del](https://redis.io/docs/latest/commands/json.del/)
- add integration tests

in redis, this command behave as below, in dice, keep consistent
`

    127.0.0.1:6379> JSON.SET user $  '{"age":13,"high":1.60,"pet":null,"language":["python","golang"], "flag":false, "partner":{"name":"tom","language":["rust"]}}'
    OK
    127.0.0.1:6379> JSON.DEL user
    (integer) 1
    127.0.0.1:6379> JSON.GET user
    (nil)
    127.0.0.1:6379> JSON.SET user $  '{"age":13,"high":1.60,"pet":null,"language":["python","golang"], "flag":false, "partner":{"name":"tom","language":["rust"]}}'
    OK
    127.0.0.1:6379> JSON.DEL user $
    (integer) 1
    127.0.0.1:6379> JSON.GET user
    (nil)
    127.0.0.1:6379> JSON.SET user $  '{"age":13,"high":1.60,"pet":null,"language":["python","golang"], "flag":false, "partner":{"name":"tom","language":["rust"]}}'
    OK
    127.0.0.1:6379> JSON.DEL user $..language
    (integer) 2
    127.0.0.1:6379> JSON.GET user
    "{\"age\":13,\"high\":1.6,\"pet\":null,\"partner\":{\"name\":\"tom\"},\"flag\":false}"
    127.0.0.1:6379> JSON.SET user $  '{"age":13,"high":1.60,"pet":null,"language":["python","golang"], "flag":false, "partner":{"name":"tom","language":["rust"]}}'
    OK
    127.0.0.1:6379> JSON.DEL user $.*
    (integer) 6
    127.0.0.1:6379> JSON.GET user
    "{}"
    127.0.0.1:6379> JSON.SET user $  '{"age":13,"high":1.60,"pet":null,"language":["python","golang"], "flag":false, "partner":{"name":"tom","language":["rust"]}}'
    OK
    127.0.0.1:6379> JSON.DEL user $.aaa
    (integer) 0
    127.0.0.1:6379> JSON.GET user
    "{\"age\":13,\"high\":1.6,\"pet\":null,\"language\":[\"python\",\"golang\"],\"flag\":false,\"partner\":{\"name\":\"tom\",\"language\":[\"rust\"]}}"
    127.0.0.1:6379> JSON.SET user $  '{}'
    OK
    127.0.0.1:6379> JSON.DEL user
    (integer) 1
    127.0.0.1:6379> JSON.GET user
    (nil)

`

### CODE REVIEW

File: core/eval.go
Comment (lucifercr07) [inline core/eval.go:1]: Shall we add integration tests as well under [this](https://github.com/DiceDB/dice/blob/master/tests/json_test.go) file?

File: core/eval.go [hunk 1]
```diff
@@ -321,6 +321,64 @@ func evalGETDEL(args []string, store *Store) []byte {
...
+		path = args[1]
+	}
+
+	// Retrieve the object from the database
+	obj := store.Get(key)
+	if obj == nil {
+		return RespZero
+	}
+
+	err := assertType(obj.TypeEncoding, ObjTypeJSON)
```
Comment (lucifercr07) [inline core/eval.go:346]: L346-L353 has similar sort of code and is used in another places as well example `JSON.CLEAR, JSON.GET` etc. May be we can extract out as function like below:
```
func assertTypeAndEncoding(typeEncoding, expectedType, expectedEncoding uint8) error {
	if err := assertType(typeEncoding, expectedType); err != nil {
		return diceerrors.NewErrWithMessage("Existing key has wrong Dice type")
	}
	if err := assertEncoding(typeEncoding, expectedEncoding); err != nil {
		return diceerrors.NewErrWithMessage("Existing key has wrong Dice type")
	}
	return nil
}
```

**Review by lucifercr07 — CHANGES_REQUESTED** [review]
LGTM, minor comments please check once.

File: core/eval.go (hunk 1 above)
Comment (jujiale) [reply core/eval.go:346]: thanks for your review, I will optimize this logic

File: core/eval.go
Comment (jujiale) [reply core/eval.go:1]: Get it. I will add

**Comment by jujiale** [comment]
made some optimize. also in json.get the redis behavior as below:
`

    127.0.0.1:6379> set key1 123 
    OK
    127.0.0.1:6379> JSON.GET key1
    (error) Existing key has wrong Redis type

`
so also made consistent with redis in json.get occur wrong type

File: core/eval.go
Comment (jujiale) [reply core/eval.go:1]: done

File: core/eval.go (hunk 1 above)
Comment (jujiale) [reply core/eval.go]: done

**Comment by JyotinderSingh** [comment]
> made some optimize. also in json.get the redis behavior as below:
> 
> `
> 
> 
> 
>     127.0.0.1:6379> set key1 123 
> 
>     OK
> 
>     127.0.0.1:6379> JSON.GET key1
> 
>     (error) Existing key has wrong Redis type
> 
> 
> 
> `
> 
> so also made consistent with redis in json.get occur wrong type

Thank you for catching and fixing this!

**Review by JyotinderSingh — APPROVED** [review]
Thanks for adding support for JSON.DEL and improving the tests @jujiale! The PR is quite well made.

Thanks for the reviews @lucifercr07! Approved.

**Pull request merged**